app/src/renderer/texture.rs

```diff
@@ -0,0 +1,42 @@
+use core::ptr::null;
+
+use crate::renderer::{
+    GRRLIB_texImg, 
+    GRRLIB_LoadTexture, 
+    GRRLIB_SetTexture
+};
+use alloc::vec::Vec;
+
+/**
+ * Our representation of a texture.
+ */
+pub struct Texture {
+    grrlib_texture: *mut GRRLIB_texImg
+}
+
+/**
+ * Implementation of texture.
+ */
+impl Texture {
+    /**
+     * Load a new texture based on PNG image data.
+     */
+    pub fn from_bytes(png_data: Vec<u8>) -> Result<Texture, &'static str> {
```

Comment:
You don't need to specify specifically as `Vec<u8>` you can use `AsRef<[u8]>`  instead which means people can embed it into program using `include_bytes`

Reply:
Ah cool, thats pretty neat, thanks :)